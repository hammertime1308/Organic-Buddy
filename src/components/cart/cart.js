import React from 'react';
import styled from 'styled-components/native';
import Icon from 'react-native-vector-icons/Ionicons';

import { COLOR } from '../../styles';

const CartContainer = styled.TouchableOpacity`
  background-color: ${COLOR.GreenHeader};
  width: 50px;
  height: 50px;
  border-radius: 50px;
  justify-content: center;
  align-items: center;
  elevation: 5;
  shadowColor: ${COLOR.Black};
  shadowOffset: { width: 0, height: 3};
`;

export const Cart = ({ onPress = () => {} }) => {
  return (
    <CartContainer activeOpacity={0.7} onPress={onPress}>
      <Icon name="ios-cart-outline" size={30} color={COLOR.White} />
    </CartContainer>
  );
};
