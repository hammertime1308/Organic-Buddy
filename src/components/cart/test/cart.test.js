import React from 'react';
import renderer from 'react-test-renderer';
import { CartIcon, CartModal } from '../cart';

describe('Cart', () => {
  it('Should render cart icon', () => {
    const tree = renderer.create(<CartIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should not render cart modal', () => {
    const tree = renderer
      .create(
        <CartModal
          visible={false}
          cart={[{ id: '1', name: 'sample name', price: 100, count: 1 }]}
          closeModal={() => {}}
          userId={0}
          clearCart={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render cart modal', () => {
    const tree = renderer
      .create(
        <CartModal
          visible={true}
          cart={[{ id: '1', name: 'sample name', price: 100, count: 1 }]}
          closeModal={() => {}}
          userId={0}
          clearCart={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
