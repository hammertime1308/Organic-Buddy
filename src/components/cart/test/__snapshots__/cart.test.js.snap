// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Should not render cart modal 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffff",
            "borderRadius": 5,
            "height": "70%",
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
            "width": "80%",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(76, 154, 42, 1)",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        Cart
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 50,
            "opacity": 1,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
            "position": "absolute",
            "right": 10,
            "top": 20,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î—‰
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "width": "100%",
              },
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  undefined,
                  Object {
                    "borderBottomWidth": 2,
                    "borderColor": "#c8e1ff",
                    "borderLeftWidth": 2,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    0,
                    Object {
                      "flexDirection": "row",
                      "overflow": "hidden",
                    },
                    Object {
                      "backgroundColor": "#f1f8ff",
                      "height": 40,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#c8e1ff",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "700",
                          "margin": 6,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    Name
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#c8e1ff",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "700",
                          "margin": 6,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    Quantity
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#c8e1ff",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "700",
                          "margin": 6,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    Price
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    0,
                    0,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      0,
                      Object {
                        "flexDirection": "row",
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#c8e1ff",
                          "borderRightWidth": 2,
                          "borderTopWidth": 2,
                        },
                        Object {
                          "justifyContent": "center",
                        },
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "margin": 6,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ]
                      }
                    >
                      sample name
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#c8e1ff",
                          "borderRightWidth": 2,
                          "borderTopWidth": 2,
                        },
                        Object {
                          "justifyContent": "center",
                        },
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "margin": 6,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ]
                      }
                    >
                      1
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#c8e1ff",
                          "borderRightWidth": 2,
                          "borderTopWidth": 2,
                        },
                        Object {
                          "justifyContent": "center",
                        },
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "margin": 6,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ]
                      }
                    >
                      100
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text
          style={
            Object {
              "color": "rgba(76, 154, 42, 1)",
              "fontSize": 20,
              "fontWeight": "700",
              "marginBottom": 10,
            }
          }
        >
          Rs. 
          100
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "green",
                  },
                ]
              }
            >
              checkout
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Cart Should render cart icon 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(76, 154, 42, 1)",
      "borderRadius": 50,
      "elevation": 5,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#000",
      "width": 50,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ffff",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î£Œ
  </Text>
</View>
`;

exports[`Cart Should render cart modal 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffff",
            "borderRadius": 5,
            "height": "70%",
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
            "width": "80%",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(76, 154, 42, 1)",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        Cart
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 50,
            "opacity": 1,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
            "position": "absolute",
            "right": 10,
            "top": 20,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î—‰
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 10,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 10,
                "width": "100%",
              },
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  undefined,
                  Object {
                    "borderBottomWidth": 2,
                    "borderColor": "#c8e1ff",
                    "borderLeftWidth": 2,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    0,
                    Object {
                      "flexDirection": "row",
                      "overflow": "hidden",
                    },
                    Object {
                      "backgroundColor": "#f1f8ff",
                      "height": 40,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#c8e1ff",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "700",
                          "margin": 6,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    Name
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#c8e1ff",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "700",
                          "margin": 6,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    Quantity
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "borderColor": "#c8e1ff",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "flex": 1,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "700",
                          "margin": 6,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ]
                    }
                  >
                    Price
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    0,
                    0,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      0,
                      Object {
                        "flexDirection": "row",
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#c8e1ff",
                          "borderRightWidth": 2,
                          "borderTopWidth": 2,
                        },
                        Object {
                          "justifyContent": "center",
                        },
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "margin": 6,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ]
                      }
                    >
                      sample name
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#c8e1ff",
                          "borderRightWidth": 2,
                          "borderTopWidth": 2,
                        },
                        Object {
                          "justifyContent": "center",
                        },
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "margin": 6,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ]
                      }
                    >
                      1
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#c8e1ff",
                          "borderRightWidth": 2,
                          "borderTopWidth": 2,
                        },
                        Object {
                          "justifyContent": "center",
                        },
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "margin": 6,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ]
                      }
                    >
                      100
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text
          style={
            Object {
              "color": "rgba(76, 154, 42, 1)",
              "fontSize": 20,
              "fontWeight": "700",
              "marginBottom": 10,
            }
          }
        >
          Rs. 
          100
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "green",
                  },
                ]
              }
            >
              checkout
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
